specs/BlockchainLayer/EpochManager/Epoch.md

```diff
@@ -0,0 +1,30 @@
+# Epoch
+
+Blocks are split into epochs.
+Genesis block is in its own epoch. After that, a block is either in its parent's epoch or
+starts a new epoch if it meets certain conditions.
+
+## BlockHeight
+Each block has a height. Genesis block has height 0, the height of each block is greater than
```

Comment:
this should go into DataStructures & primitives

Reply:
@mikhailOK looks like this comment is not addressed